libs/failsafe-controller/src/main/java/net/jodah/failsafe/SyncFailsafeConfigDelegater.java

```diff
@@ -61,13 +67,134 @@ public SyncFailsafeConfigDelegater(FailsafeConfig<R, ?> master) {
+  }
+
+  @Override
+  public SyncFailsafe<R> onAbortAsync(
+      CheckedConsumer<? extends Throwable> listener, ExecutorService executor) {
+    return super.onAbort(new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onAbortAsync(
+      ContextualResultListener<? extends R, ? extends Throwable> listener,
+      ExecutorService executor) {
+    return super.onAbort(new MonitoredContextualResultListener<>(listener, executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onCompleteAsync(
+      CheckedBiConsumer<? extends R, ? extends Throwable> listener, ExecutorService executor) {
+    return super.onComplete(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onCompleteAsync(
+      ContextualResultListener<? extends R, ? extends Throwable> listener,
+      ExecutorService executor) {
+    return super.onComplete(new MonitoredContextualResultListener<>(listener, executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailedAttemptAsync(
+      CheckedBiConsumer<? extends R, ? extends Throwable> listener, ExecutorService executor) {
+    return super.onFailedAttempt(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailedAttemptAsync(
+      CheckedConsumer<? extends Throwable> listener, ExecutorService executor) {
+    return super.onFailedAttempt(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailedAttemptAsync(
+      ContextualResultListener<? extends R, ? extends Throwable> listener,
+      ExecutorService executor) {
+    return super.onFailedAttempt(new MonitoredContextualResultListener<>(listener, executor));
+  }
+
   @Override
-  ListenerRegistry<R> registry() {
-    final ListenerRegistry<R> listenerRegistry = master.registry();
+  public SyncFailsafe<R> onFailureAsync(
+      CheckedBiConsumer<? extends R, ? extends Throwable> listener, ExecutorService executor) {
+    return super.onFailure(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailureAsync(
+      CheckedConsumer<? extends Throwable> listener, ExecutorService executor) {
+    return super.onFailure(
+        new MonitoredContextualResultListener<>(Listeners.of(listener), executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onFailureAsync(
+      ContextualResultListener<? extends R, ? extends Throwable> listener,
+      ExecutorService executor) {
+    return super.onFailure(new MonitoredContextualResultListener<>(listener, executor));
+  }
+
+  @Override
+  public SyncFailsafe<R> onRetriesExceededAsync(
+      CheckedBiConsumer<? extends R, ? extends Throwable> listener, ExecutorService executor) {
+    registry()
```

Comment:
ℹ️ There is no onRetriesExceeded() method that accepts a generic ContextualResultListener as in the other cases